src/config.c

```diff
@@ -2612,6 +2612,7 @@ static int updateClusterIp(const char **err) {
 int updateClusterHostname(const char **err) {
     UNUSED(err);
     clusterUpdateMyselfHostname();
+    clearCachedClusterSlotsResponse();
```

Comment:
shouldn't we schedule a nodes.conf update instead?

Reply:
We're not strictly required to for correctness, since it's our own hostname and we will get rebooted with the updated hostname.

Reply:
Maybe we should do the invalidate in `clusterDoBeforeSleep()` instead?

Reply:
hmm, looks like I am missing something. `clusterUpdateMyselfHostname` already schedules a nodes.conf rewrite.

![image](https://github.com/valkey-io/valkey/assets/11568491/d25eed0c-c52b-461f-96ea-db29a419a627)
![image](https://github.com/valkey-io/valkey/assets/11568491/047dab2f-e334-4199-91d4-b39acfe076e9)

Reply:
What @enjoy-binbin is saying is not conflicting with that. We schedule the rewrite, but within the multi-exec we continue to serve the stale slots response. It's not until `beforeSleep()` that we actually go and call the `configUpdateOrDie()`.

Reply:
ah, that is what I missed. Great bug. I didn't realize the importance of doing this in a multi/exec transaction.  +1 on invalidating the cache in `clusterDoBeforeSleep()` (if the `CLUSTER_TODO_SAVE_CONFIG` flag is set).

Reply:
ok, good idea, do we also want to do `CLUSTER_TODO_SAVE_CONFIG` in updateClusterAnnouncedPort (clusterUpdateMyselfAnnouncedPorts)?

Reply:
> do we also want to do CLUSTER_TODO_SAVE_CONFIG in updateClusterAnnouncedPort (clusterUpdateMyselfAnnouncedPorts)?

I think that is a good check - to avoid unnecessary invalidation